### DISCUSSION TITLE
How to disable a block?

### DISCUSSION THREAD

**Discussion by GayathriGuru** [GrapesJS/grapesjs#4182] [Q&A] [answered]
I have a custom block and I need to disable it from being dragged and dropped after it has been used once.I came across only the option to disable a block entirely when it is first added.I am looking for a situation,when it can be used only once and then the block should be disabled.Any suggestions?
My code as of now,
editor.BlockManager.add('block', 
{
      id: 'block',
      label: 'Block',
      category: 'Basic',
      //disable:true
})
and I am looking for something like,
editor.BlockManager.add('block', 
{
      id: 'block',
      label: 'Block',
      category: 'Basic',
      if(is_enabled===false)
     {
         disable:false
     }
     else
     {
          disable:true
     }
})

**Comment by ebalajier** [comment, marked as answer]
Try events 

https://grapesjs.com/docs/api/block_manager.html#available-events

**Reply by GayathriGuru** [reply]
Thank you!
